### DISCUSSION TITLE
RelationManager Tables - Attach & Detach

### DISCUSSION THREAD

**Discussion by viezel** [filamentphp/filament#82] [Ideas]
Lets use your [demo app](https://github.com/laravel-filament/demo) as an example of this feature.
When creating a new Order, you want to attach products to it.

Today you can only create a new Product, which will often not be the case. Typically all products are already created and an existing product should be able to searched and attached to the order model.

**Comment by danharrin** [comment]
Yeah, full BelongsToMany relation manager support is on the roadmap.

**Reply by ryangjchandler** [reply]
@viezel This is supported in a recent version of Filament, please upgrade and give it a go!

**Reply by viezel** [reply]
Works 👍🏻

Would be nice to be able to disable new and attach based on the state of the model. 
Eg an order that is completed should not be able to attach or remove product from

**Reply by ryangjchandler** [reply]
> Works 👍🏻
> 
> Would be nice to be able to disable new and attach based on the state of the model.
> Eg an order that is completed should not be able to attach or remove product from

@viezel You can override the `canAttach` and `canDetach` methods on the `RelationManager`. Once we've got the basic policy and gate support in #106, we can look into adding some `canCreate` and delete methods too.
